### DISCUSSION TITLE
How to configure TASMOTA for static IP on ESP8266?

### DISCUSSION THREAD

**Discussion by rtek1000** [arendst/Tasmota#14289] [Q&A] [answered]
I have two NodeMCU Dev Boards, and for some reason I'm not aware of, one of those boards failed to connect to the AP.

Both boards were able to operate as AP, but one of them could not operate as Station.

So I tried programming with Arduino. It also failed to work as a Station and connect to the AP.

After adding static IP settings the board was able to connect to the AP.

So I tried again to make the board work as Station and DHCP (no static IP) and this time it worked.

It doesn't seem to me to be a problem with my AP, but I think it's good to have this static IP option, because not everyone has access to the AP's settings to reserve an IP address.

In TASMOTA settings I did not find how to configure static IP.

**Comment by ascillato** [comment, marked as answer]
Tasmota supports setting STATIC IP ( See docs at https://tasmota.github.io/docs/Commands/#wi-fi )

**_Example:_**
To change the IP in Tasmota to 192.168.1.210, type in console:

```haskell
ipaddress1 192.168.1.210
ipaddress2 192.168.1.1
ipaddress3 255.255.255.0
restart 1
```

**Reply by rtek1000** [reply]
ipaddress1: IP
ipaddress2: Gateway
ipaddress3: Mask

For Primary DNS and Secondary DNS also have manual configuration?

**Reply by ascillato** [reply]
Yes, exactly.
As explained in the docs (link above), `ipaddress4` is DNS.

**Comment by rtek1000** [comment]
About the problem of not being able to find the IP via DHCP, it really was the AP, it happened once again.

I changed the AP and now this problem hasn't occurred.
